README.md

```diff
@@ -1,40 +1,75 @@
-make install
-```
+- [ssv-dkg](#ssv-dkg)
+  - [Table of contents:](#table-of-contents)
+  - [Goal and Introduction](#goal-and-introduction)
+    - [DKG](#dkg)
+    - [DKG tool by SSV](#dkg-tool-by-ssv)
+  - [Quick-start](#quick-start)
+    - [Overview](#overview)
+    - [Build](#build)
+    - [Operator](#operator)
+      - [Start a DKG-operator](#start-a-dkg-operator)
+      - [Launch with Docker and YAML file](#launch-with-docker-and-yaml-file)
+    - [Initiator](#initiator)
+      - [Generate Initiator identity RSA key pair](#generate-initiator-identity-rsa-key-pair)
+      - [Start DKG Initiator](#start-dkg-initiator)
+      - [Launch with Docker and YAML file](#launch-with-docker-and-yaml-file-1)
+      - [`init` message fields:](#init-message-fields)
+      - [Obtaining Operators data](#obtaining-operators-data)
+  - [Architecture](#architecture)
+    - [Flow Description:](#flow-description)
+    - [Note on DKG instance management](#note-on-dkg-instance-management)
+  - [Security notes](#security-notes)
+  - [Development Notes:](#development-notes)
+    - [TODO:](#todo)
+    - [Additional:](#additional)
+    - [Flow TODO Brakedown](#flow-todo-brakedown)
+      - [Round 1](#round-1)
+      - [Round 2](#round-2)
 
-### Operators data
 
-The data of the operators (ID, IP, Pubkey) can be collected in any way, for example a central server that you can pull the data from, or a preset file where all operators data exist.
+## Goal and Introduction
 
-Information about operators can be collected at `json` file and supplied to initiator to use for a key generation.
+### DKG
 
-Operators info file example (`./examples/operators_integration.json`):
+Distributed Key Generation is a cryptographic process that aims to solve the problem of coordinating `N` parties to cryptographically sign and verify signatures without relying on Trusted Third Parties.
+The process is demonstrated to be successful in successfully computing a key pair in the presence of a number `T` attackers in a decentralized network.
+To do so, this algorithm generates a public key, and a secret key of which no single party knows, but has some share of.
+The involvement of many parties requires Distributed key generation to ensure secrecy in the presence of malicious contributions to the key calculation.
 
-```json
-[
-  {
-    "id": 1,
-    "public_key": "LS0tLS1CRUdJTiBSU0....",
-    "ip": "http://localhost:3030"
-  },
-  {
-    "id": 2,
-    "public_key": "LS0tLS1CRUdJTiB....",
-    "ip": "http://localhost:3031"
-  }
-]
+For more information about DKG in general, [please visit this page](https://en.wikipedia.org/wiki/Distributed_key_generation).
+
+### DKG tool by SSV
+
+The SSV team built this tool leveraging [drand](https://drand.love/)'s DKG protocol implementation ([please visit their documentation](https://drand.love/docs/cryptography/#setup-phase) for more details on it). This implementation operates under the assumption of a p2p network, allowing operators to communicate.
```

Comment:
we're not using a p2p network like I mentioned, but simulate a p2p network but transmit the communication between the operators using the initiator which functions as a hub.